aten/src/ATen/native/quantized/cpu/qadd.cpp

```diff
@@ -67,9 +70,83 @@ Tensor _add_scalar_out(Tensor& out, const Tensor& self, Scalar other) {
 
 template <bool ReLUFused = false>
 class QAdd final : public c10::OperatorKernel {
+#ifdef USE_PYTORCH_QNNPACK
+Tensor qnnpack_add(Tensor qa, Tensor qb, double scale, int64_t zero_point) {
+  TORCH_CHECK(qa.ndimension() > 0, "qnnpack_add(): Got empty input tensor.");
+  Tensor qa_contig = qa.contiguous();
+  Tensor qb_contig = qb.contiguous();
+
+  const auto a_zero_point = qa_contig.q_zero_point();
+  const auto b_zero_point = qb_contig.q_zero_point();
+  const auto a_scale = qa_contig.q_scale();
+  const auto b_scale = qb_contig.q_scale();
+
+  Tensor qy = at::_empty_affine_quantized(
+      qa_contig.sizes(), at::device(kCPU).dtype(kQUInt8), scale, zero_point);
+
+  if (qa_contig.size(0) == 0) {
+    return qy;
+  }
+
+  initQNNPACK();
+
+  pytorch_qnnp_operator_t qnnpack_operator{nullptr};
+
+  size_t num_elems = qa_contig.numel() / qa_contig.size(0);
+
+  const pytorch_qnnp_status createStatus = pytorch_qnnp_create_add_nc_q8(
+      num_elems /* input size */,
+      a_zero_point /* a zero_point */,
+      a_scale /* a scale */,
+      b_zero_point /* b zero_point */,
+      b_scale /* b scale */,
+      static_cast<uint8_t>(zero_point) /* sum zero_point */,
+      scale /* sum scale */,
+      std::numeric_limits<uint8_t>::min() /* output min */,
+      std::numeric_limits<uint8_t>::max() /* output max */,
+      0 /* flags */,
+      &qnnpack_operator);
+
+  TORCH_INTERNAL_ASSERT(
+      createStatus == pytorch_qnnp_status_success,
+      "failed to create QNNPACK Add operator");
+  TORCH_INTERNAL_ASSERT(qnnpack_operator != nullptr);
```

Comment:
nit: error message? also if createStatus == success, is this assert always true?

Reply:
Yeah, this is just an additional check. Can be removed.